src/highlight.js

```diff
@@ -238,6 +241,42 @@ https://highlightjs.org/
     }
   }
 
+  function commonKeyword(word) {
+    return COMMON_KEYWORDS.indexOf(word.toLowerCase()) != -1
+  }
+
+  function scoreForKeyword(keyword, providedScore) {
+    // manual scores always win over common keywords
+    // so you can force a score of 1 if you really insist
+    if (providedScore)
+      return Number(providedScore)
+
+    return commonKeyword(keyword) ? 0 : 1;
+  }
+
+  function compileKeywords(rawKeywords, case_insensitive) {
+      var compiled_keywords = {};
+
+      var flatten = function(className, str) {
```

Comment:
`parseKeywordsAndScoresAndAddToCompiledKeywords` Yikes. LOL... 

> and move the declaration to the bottom of the function (after the return statement) so that it could be read top-down.

Does that ALWAYS work or are you relying on the NEW more recent JS hoisting changes?

Reply:
![Screen Shot 2019-10-14 at 5 04 22 PM](https://user-images.githubusercontent.com/6473/66782875-b29a1380-eea4-11e9-9722-c6c1b067614d.png)

So far not a fan because I don't see a great way to separate the return and my eyes don't find the end of the function.  is there a convention you use?

Reply:
At this point I'd considering ripping it out entirely and just giving it a 3rd argument, but then I'd need to pass the case flag also, grrrr. :-)

I really want some damn classes and proper encapsulation. :)

Reply:
If you don't like then don't do it. It's only a suggestion and you are still the maintainer, not me.

Reply:
I like trying new things... so now specific advice on how you separate the end of the function body from the helper functions?

Reply:
> `parseKeywordsAndScoresAndAddToCompiledKeywords` Yikes. LOL...

That's how a lot of the methods in our Java classes are named. Could be more readable with underscores, but in Java the convention is to use CamelCase.

Reply:
Oh yes I completely ignored your naming advice on that one... but I was trying to give the re-ordering a shot. :)  Your way IS how I would order things in Ruby... but usually I'm in "class mode" then... this codebase really messes with my head. :)

Reply:
I don't want to enforce my code-style here. But you asked for a review...

The bad thing about comments is that they easily move away from their intended location (by refactorings or added code). But a method name remains...

Reply:
We'll, you went a little deep. ;-)  I wanted more of a "anything crazy?"  But a lot of the thoughts were good... but we aren't 100% in alignment on style stuff.  Agree about comments... I didn't think I was being that terrible with the comments.  I like expressive names ButNotIfItMeansTheNamesStartToGetThislongJustTOAvoidAcomment. :-)

Reply:
I'll leave it for now (the ordering), maybe it'll grow on me... lol.

Reply:
> Does that ALWAYS work or are you relying on the NEW more recent JS hoisting changes?

As far as I know, `function abc()` is parsed before executing the code, but `var abc = function()` is parsed **while** executing the code. That's why a function in the back is still found.

I cannot pin this down to a ECMAScript-Version and I would love to have an IE6 ready right now for testing.